deep_qa/run.py

```diff
@@ -183,3 +188,62 @@ def evaluate_model(param_path: str, dataset_files: List[str]=None, model_class=N
     if dataset_files is None:
         dataset_files = model.test_files
     model.evaluate_model(dataset_files)
+
+
+def score_dataset_with_ensemble(param_paths: List[str],
+                                dataset_files: List[str],
+                                model_class=None) -> Tuple[numpy.array, numpy.array]:
+    """
+    Loads all of the models specified in ``param_paths``, uses each of them to score the dataset
+    specified by ``dataset_files``, and averages their scores, return an array of ensembled model
+    predictions.
+
+    Parameters
+    ----------
+    param_paths: List[str]
+        A list of parameter files that were used to train models.  You must have already trained
+        the corresponding model, as we'll load it and use it in an ensemble here.
+    dataset_files: List[str]
+        A list of dataset files to score, the same as you would have specified as ``test_files`` in
+        any one of the model parameter files.
+    model_class: ``DeepQaModel``, optional (default=None)
+        This option is useful if you have implemented a new model class which is not one of the
+        ones implemented in this library.
+
+    Returns
+    -------
+    predictions: numpy.array
+        Numpy array of model predictions in the format of model.outputs (typically one array, but
+        could be List[numpy.array] if your model has multiple outputs).
+    labels: numpy.array
+        The labels on the dataset, as read by the first model.  We return this so you can compute
+        whatever metrics you want, if the data was labeled.  Note that if your models all represent
+        that data differently, this will only give the first one.  Hopefully the representation of
+        the labels is consistent across the models, though; if not, the whole idea of ensembling
+        them this way is moot, anyway.
+    """
+    models = [load_model(param_path, model_class) for param_path in param_paths]
+    predictions = []
+    labels_to_return = None
+    for i, model in enumerate(models):
+        logger.info("Scoring model %d of %d", i + 1, len(models))
+        dataset = model.load_dataset_from_files(dataset_files)
+        model_predictions, labels = model.score_dataset(dataset)
+        predictions.append(model_predictions)
+        if labels_to_return is None:
+            labels_to_return = labels
+    logger.info("Averaging model predictions")
+    all_predictions = numpy.stack(predictions)
+    averaged = numpy.mean(all_predictions, axis=0)
+    return averaged, labels_to_return
```

Comment:
I'm thinking that actually this should be an `Ensembler` class, so that you can easily have options on _how_ to do the ensembling, and you can specify a parameter file for the ensemble, etc.  That should make a lot of things easier down the road, when we're integrating with the Aristo Lab stuff that aristo-eng is doing.  So I'm going to wait on this suggestion until we refactor this for a better ensembling API.